packages/app/src/app/pages/Pro/legacy-pages/WorkspacePlanSelection.tsx

```diff
@@ -186,25 +186,25 @@ export const WorkspacePlanSelection: React.FC = () => {
           {isPersonalSpace ? (
             <SubscriptionCard
               title={
-                user.subscription.plan === 'patron' ? 'Patron' : 'Personal Pro'
+                user.subscription?.plan === 'patron' ? 'Patron' : 'Personal Pro'
               }
               features={PERSONAL_FEATURES}
               cta={personalProCta}
               isHighlighted
             >
               <Stack gap={1} direction="vertical">
                 <Text size={32} weight="500">
-                  ${user.subscription.amount}
+                  ${user.subscription?.amount}
                 </Text>
-                {user.subscription.duration === 'yearly' ? (
+                {user.subscription?.duration === 'yearly' ? (
                   <Text>
                     charged annually on{' '}
-                    {format(new Date(user.subscription.since), 'MMM dd')}
+                    {format(new Date(user.subscription?.since), 'MMM dd')}
                   </Text>
                 ) : (
                   <Text>
                     charged on the{' '}
-                    {format(new Date(user.subscription.since), 'do')} of each
+                    {format(new Date(user.subscription?.since), 'do')} of each
                     month
                   </Text>
                 )}
```

Comment:
Maybe we do an early return if `subscription` does not exist? Otherwise, it will render funny things. For example, I think `new Date(undefined)` returns `Invalid Date` and then the text isn't of much help.

Reply:
You're right, thanks. I'm researching this further with the help of Julien, because a user isn't supposed to run this code at all when there is no subscription (because it had to determine that it was a legacy subscription before getting to this stage). I will wait for the outcome before merging this - as we might want to change it from `user.subscription` to the `subscription` coming from `useWorkspaceSubscription`.